tree/tree.h

```diff
@@ -15,31 +16,53 @@ class Tree{
   public:
 
     /**
-     * \brief Standard destructor.
+     * Tree class constructor
+     * \param splitter      Splitter object
+     * \param criterion     Criterion object
+     * \param sampler       Sampler object
+     * \param max_features    The maximum number of features used to create the tree.
+     * \param max_depth     The maximum depth of the tree.
+     * \param max_leaf_nodes  The maximum number of leaf nodes of the tree.
+     * \param min_samples_leaf  The minimum number of samples per leaf.
+     * \param min_samples_split The minimum number of samples per node in order to be splitted in two.
      */
-    virtual ~Tree();
+    Tree(Splitter *splitter, /*Criterion *criterion, Sampler *sampler,*/ float max_features,
+       int max_depth, int max_leaf_nodes, int min_samples_leaf, int min_samples_split);
 
     /**
-     * Builds the Decision Tree
-     * \param X              An object of Collection
-     * \param sample_weight  Weight of each sample in collection
-     * \param feature_weight Weight of each feature 
+     * \brief Standard destructor.
      */
-    virtual void Build(/*Collection X,*/
-                        std::vector<double> &sample_weight, std::vector<double> &feature_weight) = 0;
+    ~Tree();
 
     /**
-     * Predicts the labels of the instances of X based on the fitted model
-     * \return Instance labels
+     * Builds the Decision Tree.
+     * \param X        An object of Collection.
+     * \param sample_weight  Weight of each sample in collection.
+     * \param feature_weight Weight of each feature .
      */
-    virtual std::vector<int> &Predict(/*Collectiin X*/) = 0;
+    void Build(/*const Collection *X,*/
+          const std::vector<double> *sample_weight, const std::vector<double> *feature_weight);
 
+    
     /**
-     *  Returns the predicted probability distribution of each instance in X
-     *  over all possible classes.
-     *  \return Instace probability distribution
+     * Returns the leafs which each instance belongs to.
+     * \param X     An object of Collection.
+     * \param leafs Vector of leafs where the instances laid down.
      */
-    virtual std::vector<std::vector<double> > &PredictProbability(/*Collection X*/) = 0;
+    void Traverse(/*const Collection *X, std::vector<Node*> &leafs*/);
+
+  protected:
+
+    Splitter *splitter; 
+    /*Criterion *criterion;*/
+    /*Sampler *sampler;*/ 
+    float max_features;
```

Comment:
isn't it percetage of total number of features? Scikit supports integer or float. If it is integet it will use absolute number of features, if it is float it will be the percentage. I particularly prefer to use percentage. I do not have to calculate it for every dataset I let the tree do it for me.